pyproject.toml

```diff
@@ -51,10 +51,30 @@ dev = [
     "black==22.10.0",
     "interrogate==1.5.0",
     "coverage==6.5.0",
+    "ruff==0.0.290",
```

Comment:
Why the strict versioning here? Can this not be `>=`?

Reply:
Does ruff have backwards compatibility?

Its could also a similar issue with black in the sense, different versions may have slight differences but both pass which can cause odd git history

Reply:
Good point, although I don't know how stable code is with `0.0.~`. Changed.